Gemfile.lock

```diff
@@ -55,6 +56,7 @@ PLATFORMS
   x86_64-linux
 
 DEPENDENCIES
+  concurrent-ruby
```

Comment:
This is one of the most popular ruby deps, but it's runtime, so if we can avoid it that's a plus. 

It seems like it provides concurrent primitives (arrays, etc), so it's probably worth it.

Reply:
yeah, I was looking into the thread safetyness, and Ruby does provide [Mutex](https://ruby-doc.org/core-3.1.2/Thread/Mutex.html). With `concurrent-ruby` you pretty much get all the data structures in a thread-safe wrapped manner